src/components/shared/modals/modalWithInput/index.tsx

```diff
@@ -0,0 +1,118 @@
+import React, {Dispatch, SetStateAction, useEffect} from 'react';
+import {Button, Modal, Text} from 'native-base';
+import Input from 'components/shared/input';
+import styles from './styles';
+import {useForm} from 'react-hook-form';
+import {schema} from './validations';
+import {yupResolver} from '@hookform/resolvers/yup';
+import {Keyboard} from 'react-native';
+
+interface Props {
+  buttonText: string;
+  headerText: string;
+  firstInputName: string;
+  isVisible: boolean;
+  setIsVisible: Dispatch<SetStateAction<boolean>>;
```

Comment:
this is unnecessary, we should control the state from the parent component, and the modal only receives a stateful `isVisible` prop

Reply:
sounds good